src/main/java/org/broadinstitute/hellbender/utils/recalibration/BaseRecalibrationEngine.java

```diff
@@ -164,6 +165,34 @@ public static void finalizeRecalibrationTables( final RecalibrationTables tables
                 rgDatum.combine(qualDatum);
             }
         }
+
+        roundDecimalPlaces(tables);
+    }
+
+    private static void roundDecimalPlaces(final RecalibrationTables rt) {
+        for (int i = 0; i < rt.numTables(); i++) {
+            for (NestedIntegerArray.Leaf<RecalDatum> leaf : rt.getTable(i).getAllLeaves()) {
+                leaf.value.setNumMismatches(roundToNDecimalPlaces(leaf.value.getNumMismatches(), RecalUtils.NUMBER_ERRORS_DECIMAL_PLACES));
+                leaf.value.setEmpiricalQuality(roundToNDecimalPlaces(leaf.value.getEmpiricalQuality(), RecalUtils.EMPIRICAL_QUAL_DECIMAL_PLACES));
+                leaf.value.setEstimatedQReported(roundToNDecimalPlaces(leaf.value.getEstimatedQReported(), RecalUtils.EMPIRICAL_Q_REPORTED_DECIMAL_PLACES));
+            }
+        }
+    }
+
+    /**
+     * Rounds the double to the given number of decimal places.
+     * For example, rounding 3.1415926 to 3 places would give 3.142.
+     * Not put into MathUtils because it uses the horrible path of going through a formatted String (to match String.format)
+     */
+    @VisibleForTesting
+    static double roundToNDecimalPlaces(final double in, final int n) {
+        if (n < 1) {
+            throw new IllegalArgumentException("cannot round to " + n + " decimal places");
+        }
+
+        //Note: this code is very ugly because it goes through parsing and unparsing.
+        //The reason is that we want to match what writing the recalibration tables down to a file would do.
+        return Double.valueOf(String.format("%." + n + "f", in));
```

Comment:
Was surprised to learn that `BigDecimal` was not working with `RoundingMode.HALF_UP`, so did a little digging and found this: http://stackoverflow.com/questions/7124448/how-does-java-math-roundingmode-work

Basically if you replace `new BigDecimal(in)` with `BigDecimal.valueOf(in)` it works. Ie., 

           BigDecimal bd = BigDecimal.valueOf(1.025);
           bd = bd.setScale(2, RoundingMode.HALF_UP);
           System.out.println(bd.doubleValue());

prints 1.03